nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ExtractEmailAttachments.java

```diff
@@ -0,0 +1,201 @@
+    public static final String ATTACHMENT_ORIGINAL_UUID = "email.attachment.parent.uuid";
+
+    public static final Relationship REL_ATTACHMENTS = new Relationship.Builder()
+            .name("attachments")
+            .description("Each individual attachment will be routed to the attachments relationship")
+            .build();
+    public static final Relationship REL_ORIGINAL = new Relationship.Builder()
+            .name("original")
+            .description("The original file")
+            .build();
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("Flowfiles that could not be parsed")
+            .build();
+    private Set<Relationship> relationships;
+    private List<PropertyDescriptor> descriptors;
+
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_ATTACHMENTS);
+        relationships.add(REL_ORIGINAL);
+        relationships.add(REL_FAILURE);
+        this.relationships = Collections.unmodifiableSet(relationships);
+
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+
+        this.descriptors = Collections.unmodifiableList(descriptors);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+        final ComponentLog logger = getLogger();
+        final FlowFile originalFlowFile = session.get();
+        if (originalFlowFile == null) {
+            return;
+        }
+        final List<FlowFile> attachmentsList = new ArrayList<>();
+        final List<FlowFile> invalidFlowFilesList = new ArrayList<>();
+        final List<FlowFile> originalFlowFilesList = new ArrayList<>();
+
+        session.read(originalFlowFile, new InputStreamCallback() {
+                @Override
+                public void process(final InputStream rawIn) throws IOException {
+                    try (final InputStream in = new BufferedInputStream(rawIn)) {
+                        Properties props = new Properties();
+                        Session mailSession = Session.getDefaultInstance(props, null);
+                        MimeMessage originalMessage = new MimeMessage(mailSession, in);
+                        MimeMessageParser parser = new MimeMessageParser(originalMessage).parse();
+                        // RFC-2822 determines that a message must have a "From:" header
+                        // if a message lacks the field, it is flagged as invalid
+                        Address[] from = originalMessage.getFrom();
+                        Date sentDate = originalMessage.getSentDate();
+                        if (from == null || sentDate == null ) {
+                            // See ya later my friend...
+                            throw new MessagingException("Message failed RFC2822 validation");
```

Comment:
Apologies but I am not sure I understand what you mean?

Logic as I read is: if message fails parsing, `MessagingException `is raised and captured via `catch `and added to `invalidFlowFileList`. Code continues until `session.transfer(invalidFlowFilesList, REL_FAILURE);`